### PR TITLE
#166003010 Authentication private routes 

### PR DESCRIPTION
**Pull request by dannylwe**
#### What does this PR do?
Enable social login

#### Description of Task to be completed?
This PR introduces private routes which ensure only authenticated users can access
specific routes

#### How should this be manually tested?
run `git fetch authentication-private-routes-166003010`
run `git checkout authentication-private-routes-166003010`
run `npm install`
run `npm run start:dev`

Go to route `/profile` or `/editprofile` without log in, automatically rerouted to `/` which is our dashboard. After authentication all routes are accessible.

#### What are the relevant pivotal tracker stories?
[#166003010](https://www.pivotaltracker.com/story/show/166003010)

### CODE REVIEW

File: src/pages/Routes.jsx [hunk 1]
```diff
@@ -1,27 +1,44 @@
...
-        <Route exact path="/" component={LandingComponent} />
-        <Route path="/register" component={SignupComponent} />
-        <Route path="/login" component={SigninComponent} />
-        <Route path="/password-reset/" component={ResetPassword} />
-        <Route path="/reset/:token/change/" component={ChangePassword} />
-      </Switch>
-    </Router>
-  </>
-);
+class Routes extends Component {
```
Comment (ezrogha) [inline src/pages/Routes.jsx:14]: Why did you convert this component to a state component?

**Review by ezrogha — COMMENTED** [review]
Hey @dannylwe good work with the PR.

File: src/pages/Routes.jsx (hunk 1 above)
Comment (dannylwe) [reply src/pages/Routes.jsx:14]: @ezrogha  I was setting the state in react and checking against it in localstorage

**Review by felixkiryowa — COMMENTED** [review]
@dannylwe  please change the commit because , the changes being reflected is @ezrogha work

File: src/actions/profileActions.js [hunk 2]
```diff
@@ -1,20 +1,18 @@
...
-    .then((response) => {
-      dispatch({ type: USER, payload: response.data.profile });
-      localStorage.setItem('username', response.data.user.username);
-    })
-    .catch(() => {
-      dispatch({
-        type: ERROR,
-        payload: 'Can not fetch your data, please login again',
-      });
+export const fetchProfileActions = username => dispatch => axios
```
Comment (hadijahkyampeire) [inline src/actions/profileActions.js:4]: why do you have these changes

File: src/actions/signupSigninActions.js [hunk 3]
```diff
@@ -1,19 +1,17 @@
...
-      notify.show('Registration successful', 'success', 4000);
-      history.push('/login');
-    })
-    .catch((error) => {
-      const errorObject = error.response.data.errors;
-      const errorMessage = errorObject[Object.keys(errorObject)[0]][0];
-      notify.show(errorMessage, 'error', 4000);
-    });
-};
+export const signUpUser = (user, history) => () => axios
```
Comment (hadijahkyampeire) [inline src/actions/signupSigninActions.js:4]: same here that means you did not rebase well

File: src/actions/profileActions.js (hunk 2 above)
Comment (dannylwe) [reply src/actions/profileActions.js:4]: @hadijahkyampeire these changes are from eslint

File: src/actions/signupSigninActions.js (hunk 3 above)
Comment (dannylwe) [reply src/actions/signupSigninActions.js:4]: @hadijahkyampeire these changes are from eslint. It complains otherwise

**Review by hadijahkyampeire — APPROVED** [review]

**Pull request merged**